src/parameters/definition_types.ts

```diff
@@ -145,6 +145,7 @@ interface NumberParameterDefinition extends BaseParameterDefinition<number> {
 interface OAuth2ParameterDefinition extends BaseParameterDefinition<string> {
   type: typeof SlackPrimitiveTypes.oauth2;
   oauth2_provider_key: string;
+  require_end_user_auth: boolean;
```

Comment:
Is this a required property? That is, must users specify it?

Reply:
It's an optional property because it's not specified as required in the `oauth2` type definition in [Webapp](https://slack-github.com/slack/webapp/pull/184865/files#diff-9cf257daf9d0ce35304e8c43f85a59a5b99c72e734106d44e506766b10a976b2R3)

Reply:
Would be nice to add a JSDoc above the property as well, explaining what this option does to the developer.

Reply:
Sure, I can do both! Re. making the property optional in the SDK - I wasn't able to see a difference in behavior when it was marked optional vs. not. Is there anyway to tell whether the optionality is being respected when I code up the function?

Reply:
The fact you saw no behavioural difference is probably an indication we aren't doing a very good job validating the structure of these parameter definitions on the SDK side, which is something we will work on 😬 But, even if it's not working right now, once we get to it, it helps me remember this conversation that indeed, this particular property can be optional and doesn't have to exit 😅